### PR TITLE
Remove Type suffix from some enums.

### PR DESCRIPTION
**Pull request by katmsft**
# Pull Request Checklist

Please leverage this checklist as a reminder to address commonly occurring feedback when submitting a pull request to make sure your PR can be reviewed quickly:

See the detailed list in the [contributing guide](https://github.com/Azure/azure-sdk-for-cpp/blob/master/CONTRIBUTING.md#pull-requests).

- [x] [C++ Guidelines](https://azure.github.io/azure-sdk/cpp_introduction.html)
- [x] Doxygen docs
- [x] Unit tests
- [x] No unwanted commits/changes
- [x] Descriptive title/description
  - [x] PR is single purpose
  - [x] Related issue listed
- [x] Comments in source
- [x] No typos
- [x] Update changelog
- [x] Not work-in-progress
- [x] External references or docs updated
- [x] Self review of PR done
- [x] Any breaking changes?

### CODE REVIEW

**Comment by Jinming-Hu** [comment]
Where is `DataLake::Models::PathExpiryOptions` used?

**Review by Jinming-Hu — CHANGES_REQUESTED** [review]
see in comment

**Comment by Jinming-Hu** [comment]
@danieljurek CI seems to be down, can you take a look?

**Review by Jinming-Hu — APPROVED** [review]

**Comment by danieljurek** [comment]
@Jinming-Hu, @katmsft  -- Thanks for re-running the pipeline. This was caused by an issue on the Engineering System side that we have resolved. Re-running the pipeline should succeed now.

**Comment by ahsonkhan** [comment]
@danieljurek new intermittent infra failure on MacOS during test resource deployment:
https://dev.azure.com/azure-sdk/internal/_build/results?buildId=796737&view=logs&j=267db408-59fc-59d9-cfd0-fd1515cbe21c&t=ef2413d8-cecc-5056-5cea-ca63a0346a44&l=10357
```text
Write-Error: /Users/runner/work/1/s/eng/common/TestResources/New-TestResources.ps1:412
Line |
 412 |          $deployment = Retry {
     |                        ~~~~~~~
     | Exception calling "Invoke" with "0" argument(s): "The running
     | command stopped because the preference variable
     | "ErrorActionPreference" or common parameter is set to Stop:
     | 9:58:53 PM - The deployment 'tb8792f0a8ce54c5e' failed with
     | error(s). Showing 1 out of 1 error(s). Status Message: A hash
     | conflict was encountered for the role Assignment ID. Please
     | use a new Guid.
     | (Code:HashConflictOnDifferentRoleAssignmentIds) 
     | CorrelationId: 6499df0b-227d-4129-bb0e-71b940b1c7d2
```

File: sdk/storage/azure-storage-files-shares/CHANGELOG.md [hunk 1]
```diff
@@ -29,7 +29,11 @@
 - Renamed `ShareRetentionPolicy` to `RetentionPolicy`.
 - Renamed `ShareProtocolSettings` to `ProtocolSettings`.
 - Renamed `CopyStatusType` to `CopyStatus`
-- Removed `FileRangeWriteType`, `ShareFileRangeList` and `ShareStats`.
+- Removed `FileRangeWriteType`, `ShareFileRangeList`, `FileRangeWriteFromUrlType`, `FileRange`, `ClearRange`, `SharePermission`, `LeaseAction` and `ShareStats`.
+- Renamed `LeaseDurationType` to `LeaseDuration`, `LeaseStateType` to `LeaseState` and `LeaseStatusType` to `LeaseStatus`.
+- Renamed `ListSharesIncludeType` to `ListSharesIncludeFlags`.
```
Comment (ahsonkhan) [inline sdk/storage/azure-storage-files-shares/CHANGELOG.md:34]: For enums, when do we decide to use the "Flags" suffix, vs not?

**Review by ahsonkhan — APPROVED** [review]

**Comment by danieljurek** [comment]
Thanks @ahsonkhan, we've seen this issue before here: https://github.com/Azure/azure-sdk-tools/issues/1240 ... It seems to show up from time to time. 

The template is well formed to create unique assignment names and while re-running will likely fix the issue there may be some more we can do to help fix this. I'm not sure if the role assignment needs to be unique to the resource, group, subscription, or at some other level (like globally).

**Pull request merged**